<launch>

  <!-- Detect Ball Parameters-->
  <arg name="h_min_forward" default="0" />
  <arg name="h_max_forward" default="40" />
  <arg name="s_min_forward" default="52" />
  <arg name="s_max_forward" default="255" />
  <arg name="v_min_forward" default="174" />
  <arg name="v_max_forward" default="255" />
  <param name="/detect_ball_forward/h_min" value="$(arg h_min_forward)" />
  <param name="/detect_ball_forward/h_max" value="$(arg h_max_forward)" />
  <param name="/detect_ball_forward/s_min" value="$(arg s_min_forward)" />
  <param name="/detect_ball_forward/s_max" value="$(arg s_max_forward)" />
  <param name="/detect_ball_forward/v_min" value="$(arg v_min_forward)" />
  <param name="/detect_ball_forward/v_max" value="$(arg v_max_forward)" />



  <node name="detect_ball_forward" pkg="coconuts_odroid" type="detect_ball_forward" />

  <!--stereo image proc - forward camera-->
  <node ns="camera_forward" name= "image_proc" pkg="image_proc" type="image_proc"/>

  <!-- forward camera uses usb_cam -->
  <node name= "camera_forward" pkg="usb_cam" type="usb_cam_node" >
    <param name="video_device" type="string" value="/dev/video2"/>
    <param name="pixel_format" type="string" value="yuyv"/>
    <param name="framerate" type="int" value="5"/>
    <param name="camera_frame_id" type="string" value="camera_forward" /> 
    <param name="camera_info_url" type="string" value="file://$(find coconuts_odroid)/src/camera_forward_calibration.yaml"/>
  </node>

  <!-- upward camera uses uvc_camera -->
  <node ns="camera_up" name="camera_up" pkg="uvc_camera" type="uvc_camera_node" >
    <param name="width" type="int" value="160" />
    <param name="height" type="int" value="120" />
    <param name="brightness" type="int" value="150"/>
    <param name="fps" type="int" value="30" />
    <param name="frame" type="string" value="camera_up" />
    <param name="device" type="string" value="/dev/video1" />
    <param name="auto_focus" type="bool" value="0" />
    <param name="camera_frame_id" type="string" value="camera_up" /> 
    <param name="camera_info_url" type="string" value="file://$(find coconuts_odroid)/src/camera_up_calibration.yaml"/>
    </node>


  <!-- locate ball publishes ball transform in 3d space -->
  <node name="locate_ball_forward" pkg="coconuts_odroid" type="locate_ball_forward" output="screen" />

  <!-- detect ball down parameters-->
  <arg name="h_min_down" default="0" />
  <arg name="h_max_down" default="55" />
  <arg name="s_min_down" default="0" />
  <arg name="s_max_down" default="255" />
  <arg name="v_min_down" default="195" />
  <arg name="v_max_down" default="255" />
  <param name="/detect_ball_down/h_min" value="$(arg h_min_down)" />
  <param name="/detect_ball_down/h_max" value="$(arg h_max_down)" />
  <param name="/detect_ball_down/s_min" value="$(arg s_min_down)" />
  <param name="/detect_ball_down/s_max" value="$(arg s_max_down)" />
  <param name="/detect_ball_down/v_min" value="$(arg v_min_down)" />
  <param name="/detect_ball_down/v_max" value="$(arg v_max_down)" />

  <node name="detect_ball_down" pkg="coconuts_odroid" type="detect_ball_down" />
  <!-- downward camera uses usb_cam, this one is not on the USB bus -->
  <node name= "camera_down" pkg="usb_cam" type="usb_cam_node" >
    <param name="video_device" type="string" value="/dev/video0"/>
    <param name="framerate" type="int" value="5"/>
    <param name="pixel_format" type="string" value="mjpeg"/>
    <param name="camera_frame_id" type="string" value="camera_down" /> 
    <param name="camera_info_url" type="string" value="file://$(find coconuts_odroid)/src/camera_down_calibration.yaml"/>
  </node> 

 <node name="globalLocalizationPY" pkg="coconuts_odroid" type="globalLocalizationPY.py" />



  </launch>

